ra2ce/runners/damages_analysis_runner.py

```diff
@@ -66,103 +64,82 @@ def _get_result_link_based(
         analysis_config: AnalysisConfigWrapper,
         result_segment_based: gpd.GeoDataFrame,
     ) -> gpd.GeoDataFrame:
-        def get_damage_result_columns(
-            damage_curve: DamageCurveEnum,
-            event: str,
-            result_segment_based: gpd.GeoDataFrame,
-        ) -> list[str]:
-            if (
-                damage_curve == DamageCurveEnum.HZ
-            ):  # Replace with the actual enum or constant if needed
+        # Step 00: define parameters
+        base_graph_hazard_graph = analysis_config.graph_files.base_graph_hazard.graph
+        damage_curve = analysis.analysis.damage_curve.name
+        segment_id_column = "rfid_c"
+
+        # Find the hazard columns; these may be events or return periods
+        event_cols = [
+            col
+            for col in result_segment_based.columns
+            if (col[0].isupper() and col[1] == "_")
+        ]
+        events = set([x.split("_")[1] for x in event_cols])  # set of unique events
+
+        # Step 0: create a deep copy of the base_graph_hazard to compose further as the final outcome of this process
+        damages_link_based_graph = base_graph_hazard_graph.copy()
+
+        # Step 1: Create a new attribute damage_segments_list for each edge
+        for event in events:
+            if damage_curve == analysis.analysis.damage_curve.HZ.name:
```

Comment:
I thought the idea was to do here a comparison by  enums?

Reply:
Agree. That was given as a comment, but not fully implemented. I'll change it.